projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_connection_base.py

```diff
@@ -157,6 +158,31 @@ def execute_query(self, query: str) -> List[List[Any]]:
                 List[List[Any]], res
             )  # we return None in case of DML. This is not PEP249 compliant, but is more convenient
 
+    def read_sql_as_dataframe(self, query: str) -> pd.DataFrame:
+        """Read SQL statement and returns the result as pandas DataFrame"""
+        with self.connect() as db:
+            try:
+                res = pd.read_sql(query, con=db)
```

Comment:
I would probably just do `return pd.read_sql(query, con=db)` here and then `return None` at the end of the catch statement. But might just be personal preference.